models/articles.js

```diff
@@ -2,15 +2,61 @@
 const db = require("../db/connection");
 
 // Model Funcs
-exports.selectArticles = async () => {
-  const result = await db.query(
-    `SELECT articles.*, COUNT(comments.article_id)::INTEGER 
-    AS comment_count
-    FROM comments
-    RIGHT JOIN articles
-    ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id;`
-  );
+exports.selectArticles = async (
+  sort_by = "created_at",
+  order = "DESC",
+  topic
+) => {
+  // First off defining our valid coloumn and order queries, as well as setting up an array for potential query values.
+  const validColumns = [
+    "article_id",
+    "title",
+    "topic",
+    "author",
+    "body",
+    "created_at",
+    "votes",
+    "comment_count",
+  ];
+  const validOrders = ["ASC", "DESC"];
+  const validTopics = [
```

Comment:
Technically, this is good. I can see you've got topics from both the dev and test databases in here. But I wonder... how might you go about getting only the topics that are relevant to the database that you're in? Consider this a stretch goal of sorts!

Reply:
Thanks Joey, yeah I did think hard-coding in all the topics wasn't the 'best' tactic to go for. I will have a think and try to implement something that checks the relevant topics!